src/lib.rs

```diff
@@ -16,11 +16,14 @@
 //! result may be poisoned on panics. The poison is propagated from that of the underlying `lock()`
 //! method, so for `RwLock`s, the same rule applies for when a lock may be poisioned.
 
-use std::rc;
-use std::sync;
 use std::ops::Deref;
 use std::ops::DerefMut;
+use std::rc;
+use std::sync;
 
+// Each structure holds the guard in an Option to ensure that we drop the guard before we drop the
```

Comment:
Ah, sorry, I phrased my comment poorly. I meant for this to be added _to_ the ATTENTION READERS section, not above it :)

Reply:
My bad! I followed the comment too literally :)